docs/topic-guides/system-concepts/scheduling.txt

```diff
@@ -119,36 +119,55 @@ enabled:
- Gang-scheduling on Kubernetes
-*******************************
-
-Kubernetes does not natively support gang-scheduling and its default
-scheduler will schedule pods on a first-come-first-serve basis. This
-approach is problematic if a user submits several multi-pod jobs at
-once; first-come-first-serve could result in a cluster deadlock.
-Determined is able to support gang-scheduling by using the lightweight
-coscheduling plugin, which extends the Kubernetes scheduler and blocks
-scheduling of pods unless there are enough resources for all the pods in
-the job. To function, the plugin requires special labels to be set that
-specify the amount of nodes that each job needs for execution.
-Determined is able to automatically calculate and set these labels for
-its experiments.
-
-Importantly, the coscheduling plugin does not work with cluster
-autoscaling. Static node pools must be used to achieve gang-scheduling.
-Also, while the plugin allocates resources to jobs based on their
-priority, it does not support preemption. Any low priority task will be
-able to finish before a higher priority task can begin running.
-Additionally, there isn't an implementation of ``max_slots`` or
-``max_concurrent_trials`` that would limit the resources of an
-experiment, i.e. one for hyperparameter search. Lastly, Determined's
-capability to automatically set pod labels is restricted to GPU
-experiments; it is unable to do the same for CPU experiments or user
-commands. If gang-scheduling is desired for these, it must be set
-manually via the environment field in the config. For instance:
-
-.. code::
+.. _scheduling-on-kubernetes:
+
+**************************
+ Scheduling on Kubernetes
+**************************
+
+By default, Kubernetes schedules pods one at a time, in a
+first-come-first-serve (FCFS) manner. This can result in problematic
+behavior for deep learning workloads, particularly for distributed
+training jobs that use many GPUs. A distributed training job that uses
+multiple pods requires all pods to be scheduled and running in order to
+make progress. Because Kubernetes does not support gang scheduling by
+default, cluster deadlocks can arise. For example, suppose that two
+experiments are launched simultaneously that each use 16 GPUs on a
+cluster with only 16 GPUs. It is possible that Kubernetes will assign
+some GPUs to one experiment and some GPUs to the other. Because neither
+experiment will receive the resources it needs to begin executing, the
+system will wait indefinitely. Kubernetes also does not support
+priority-based scheduling.
+
+Determined addresses these problems through the use of the `lightweight
+coscheduling plugin
+<https://github.com/kubernetes-sigs/scheduler-plugins/tree/release-1.18/pkg/coscheduling>`__,
+which extends the Kubernetes scheduler to support gang scheduling and
+priority-based scheduling. To implement gang scheduling, the
+coscheduling will not schedule a pod if there are not enough resources
```

Comment:
Suggestion: change "coscheduling" to "coscheduling plugin", or "plugin", or "coscheduler"